kore/src/Kore/Log/ErrorRewriteLoop.hs

```diff
@@ -27,23 +38,32 @@ import Kore.Syntax.Variable
 
 import Log
 
-newtype ErrorRewriteLoop =
-    ErrorRewriteLoop { getRule :: RewriteRule Variable }
+data ErrorRewriteLoop =
+    ErrorRewriteLoop
+        { rule :: RewriteRule Variable
+        , errorCallStack :: CallStack
+        }
+    deriving (Show)
+
+instance Exception ErrorRewriteLoop
 
 instance Pretty ErrorRewriteLoop where
-    pretty errorRewriteRule =
-        Pretty.vsep
+    pretty ErrorRewriteLoop { rule, errorCallStack } =
+        Pretty.vsep $
             [ "Found semantic rule with the same left- and right-hand side at:"
             , Pretty.pretty
-                . sourceLocation . attributes . getRewriteRule . getRule
-                $ errorRewriteRule
+                . sourceLocation . attributes . getRewriteRule $ rule
             , "Execution would not terminate when the rule applies."
+            , "Aborting execution."
```

Comment:
We can remove this. It was only there because I didn't know what else to pass to `error`. :smile: 

```suggestion
```